BUILDING.md

```diff
@@ -53,66 +29,169 @@ following packages are installed:
 
-> Warning: On macOS, by default, CMake prefers Framework installations
+> **Warning**
+>
+> On macOS, by default, CMake prefers Framework installations
 > (see: [$CMAKE_FIND_FRAMEWORK](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_FRAMEWORK.html)).
 > This means dependencies such as Python may default to the system
 > installation despite alternatives being present. In addition, if you
 > are attempting to use the ConanCenter cpython package as an
 > alternative, we have encountered missing header issues that may
 > require additional work/configuration to resolve.
 
-## Building with CMake
-
-We use the CMake build system for compiling the C++ core library and
-its Python bindings.
-
-By default, the project will build shared libraries and set the
-installation directory to `dist` under the build directory. This can be
-overridden by setting the built-in CMake variables `BUILD_SHARED_LIBS`
-and `CMAKE_INSTALL_PREFIX`, respectively.
+## Building
 
-If shared library builds are disabled, then the core library will be
-built and installed as a static library, and statically linked into the
-Python module.
+OpenAssetIO consists of a core C++ project, as well as a set of C and
+Python bindings to that core libary. These modules are organized into a
+CMake project, and are all built together by default.
 
-In order to build and install the binary artifacts and Python sources,
-assuming the current working directory is at the repository root, run
+Assuming your build environment has the required dependencies,
+[(See Above)](#build-dependencies), OpenAssetIO can be built and installed by
+running :
 
 ```shell
 cmake -S . -B build
 cmake --build build
 cmake --install build
 ```
 
-The generated install tree can then be used directly (e.g. by extending
+> **Note**
+>
+> A common error to encounter is to find that CMake cannot resolve
+> its dependencies via [`find_package`](https://cmake.org/cmake/help/latest/command/find_package.html)
+>
+> A convenient way to solve this is to use the [`CMAKE_TOOLCHAIN_FILE`](https://cmake.org/cmake/help/v3.24/envvar/CMAKE_TOOLCHAIN_FILE.html)
+> environment variable to augment CMake's package search paths.
+>
+> For example, to use [`conan`](https://docs.conan.io/en/1.46/installation.html)
+> to resolve dependencies, (as we do during internal development) run
+> the following commands from the root of the repository prior to
+> building :
+>
+> ```shell
+> conan install -if .conan resources/build --build=missing
+> export CMAKE_TOOLCHAIN_FILE=`pwd`/.conan/conan_paths.cmake
+> ```
+>
+> (These commands assume a Linux host, translate as necessary).
+>
+> This installs OpenAssetIO's dependencies as described in
+> `/resources/build/conanfile.py`, and allows CMake to deduce the
+> locations via referencing the generated `conan_paths.cmake`.
+
+The modules are installed to `/build/dist`. This default location can
+be overridden by setting `CMAKE_INSTALL_PREFIX` prior to build.
```

Comment:
[consider][minor] Or by passing `--install-prefix` to `cmake` at configure time.  We probably don't want to confuse things by adding several alternatives, though.

Reply:
Aye, I don't think one can be said to be better than the other, and better not to introduce branching options.